### PR TITLE
Added documentation for secret length minimums to use shared secret JWT HS Algorithms

### PR DESCRIPTION
**Pull request by tannernicol-okta**
## Description:
- Added documentation for secret length minimums to use shared secret JWT HS Algorithms
- PR related to Nimbus Library Upgrade (scheduled for production in October)

### Resolves:

* [OKTA-312145](https://oktainc.atlassian.net/browse/OKTA-312145)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/api/oidc/index.md [hunk 1]
```diff
@@ -1397,6 +1397,8 @@ If you configured your client to use the `client_secret_jwt` client authenticati
 
 Provide the `client_id` in a JWT that you sign with the `client_secret` using an HMAC SHA algorithm (HS256, HS384, or HS512). The JWT must also contain other values, such as issuer and subject. See [Token claims for client authentication with client secret or private key JWT](/docs/reference/api/oidc/#token-claims-for-client-authentication-with-client-secret-or-private-key-jwt).
 
+> **Note:** JWT with a shared key requires a secret of at least 32 characters to satisfy HS256 cryptographic minimums. Clients that attempt to set token_endpoint_auth_method = `client_secret_jwt` with an imported secret less than 32 characters will receive a validation error. Clients that send Okta a JWT signed with HS256, HS384, HS512 to verify with a secret less than 32 characters will receive the error code `oauth2.error.code.client_assertion_invalid_secret_length`
```
Comment (jakubvul-okta) [inline packages/@okta/vuepress-site/docs/reference/api/oidc/index.md]:
```suggestion
> **Note:** JWTs with a shared key require a secret that is at least 32 characters in length to satisfy HS256 cryptographic minimums. Clients that attempt to set `token_endpoint_auth_method` to `client_secret_jwt` with an imported secret less than 32 characters will receive a validation error. Clients that send Okta a JWT signed with HS256, HS384, or HS512 to verify with a secret less than 32 characters will receive this error: `oauth2.error.code.client_assertion_invalid_secret_length`
```

File: packages/@okta/vuepress-site/docs/reference/api/oidc/index.md [hunk 2]
```diff
@@ -1397,6 +1397,8 @@ If you configured your client to use the `client_secret_jwt` client authenticati
 
 Provide the `client_id` in a JWT that you sign with the `client_secret` using an HMAC SHA algorithm (HS256, HS384, or HS512). The JWT must also contain other values, such as issuer and subject. See [Token claims for client authentication with client secret or private key JWT](/docs/reference/api/oidc/#token-claims-for-client-authentication-with-client-secret-or-private-key-jwt).
 
+> **Note:** JWT with a shared key requires a secret of at least 32 characters to satisfy HS256 cryptographic minimums. Clients that attempt to set token_endpoint_auth_method = `client_secret_jwt` with an imported secret less than 32 characters will receive a validation error. Clients that send Okta a JWT signed with HS256, HS384, HS512 to verify with a secret less than 32 characters will receive the error code `oauth2.error.code.client_assertion_invalid_secret_length`.
```
Comment (jakubvul-okta) [inline packages/@okta/vuepress-site/docs/reference/api/oidc/index.md:1400]:
```suggestion
> **Note:** JWTs with a shared key require a secret that is at least 32 characters in length to satisfy HS256 cryptographic minimums. Clients that attempt to set `token_endpoint_auth_method` to `client_secret_jwt` with an imported secret less than 32 characters will receive a validation error. Clients that send Okta a JWT for verification signed with HS256, HS384, or HS512 with a secret less than 32 characters will receive an error: `oauth2.error.code.client_assertion_invalid_secret_length`.
```

**Review by jakubvul-okta — APPROVED** [review]

**Review by ravidontharaju-okta — APPROVED** [review]

File: packages/@okta/vuepress-site/docs/reference/api/oidc/index.md [hunk 3]
```diff
@@ -1397,6 +1397,8 @@ If you configured your client to use the `client_secret_jwt` client authenticati
 
 Provide the `client_id` in a JWT that you sign with the `client_secret` using an HMAC SHA algorithm (HS256, HS384, or HS512). The JWT must also contain other values, such as issuer and subject. See [Token claims for client authentication with client secret or private key JWT](/docs/reference/api/oidc/#token-claims-for-client-authentication-with-client-secret-or-private-key-jwt).
 
+> **Note:** JWTs with a shared key require a secret that is at least 32 characters in length to satisfy HS256 cryptographic minimums. Clients that attempt to set `token_endpoint_auth_method` to `client_secret_jwt` with an imported secret less than 32 characters will receive a validation error. Clients that send Okta a JWT for verification signed with HS256, HS384, or HS512 with a secret less than 32 characters will receive an error: `oauth2.error.code.client_assertion_invalid_secret_length`.
```
Comment (danielzeller-okta) [inline packages/@okta/vuepress-site/docs/reference/api/oidc/index.md:1400]: `oauth2.error.code.client_assertion_invalid_secret_length` is not the error the client will receive, that is the localization string

**Review by danielzeller-okta — CHANGES_REQUESTED** [review]
please fix the error message

**Review by danielzeller-okta — APPROVED** [review]
LGTM 🚢

File: packages/@okta/vuepress-site/docs/reference/api/oidc/index.md (hunk 3 above)
Comment (varrunramani-okta) [reply packages/@okta/vuepress-site/docs/reference/api/oidc/index.md]: 1. Should `HS256` , `HS384` and `HS512` be in back-ticks?
2. Do we have say "will receive a validation error", I think just expected behavior should be sufficient. It is understood that if clients do not meet those requirements, they will have a validation error. 
3. Not sure if we need to talk about both imported secrets and signatures, given both are requirements for using a `HS*` algorithm?

**Proposal**:
Client Secret JWTs using `HS256` , `HS384` or `HS512` schemes require a secret length of at least 32 characters to satisfy minimum cryptographic requirements.

**Review by kevinsee-okta — APPROVED** [review]

**Pull request merged**